include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -288,10 +288,48 @@ namespace internal
 
           univariate_shape_data.nodal_at_cell_boundaries = true;
 
-          lexicographic_numbering.resize(n_dofs);
+          std::vector<unsigned int> scalar_lexicographic(n_dofs);
 
           for (unsigned int i = 0; i < n_dofs; ++i)
-            lexicographic_numbering[i] = i;
+            scalar_lexicographic[i] = i;
+
+
+          if (fe_in.n_components() == 1)
+            lexicographic_numbering = scalar_lexicographic;
+          else
+            {
+              // have more than one component, get the inverse
+              // permutation, invert it, sort the components one after one,
+              // and invert back
+              std::vector<unsigned int> scalar_inv =
+                Utilities::invert_permutation(scalar_lexicographic);
+              std::vector<unsigned int> lexicographic(
+                fe_in.n_dofs_per_cell(), numbers::invalid_unsigned_int);
+              unsigned int components_before = 0;
+              for (unsigned int e = 0; e < base_element_number; ++e)
+                components_before += fe_in.element_multiplicity(e);
+              for (unsigned int comp = 0;
+                   comp < fe_in.element_multiplicity(base_element_number);
+                   ++comp)
+                for (unsigned int i = 0; i < scalar_inv.size(); ++i)
+                  lexicographic[fe_in.component_to_system_index(
+                    comp + components_before, i)] =
+                    scalar_inv.size() * comp + scalar_inv[i];
+
+              // invert numbering again. Need to do it manually because we might
+              // have undefined blocks
+              lexicographic_numbering.resize(fe_in.element_multiplicity(
+                                               base_element_number) *
+                                               fe->n_dofs_per_cell(),
+                                             numbers::invalid_unsigned_int);
+              for (unsigned int i = 0; i < lexicographic.size(); ++i)
+                if (lexicographic[i] != numbers::invalid_unsigned_int)
+                  {
+                    AssertIndexRange(lexicographic[i],
+                                     lexicographic_numbering.size());
+                    lexicographic_numbering[lexicographic[i]] = i;
+                  }
+            }
```

Comment:
I have modified a test so that it uses the project function. While doing this, I notices that `ShapeInfo::lexicographic_numbering` has not been set up properly. I have not noticed this till now because there a no tests using multiple components till now...

Reply:
@kronbichler I am wondering how to proceed here. Should I create a separate PR. I guess we should create an internal function for setting of this data structure because it is a copy from the other code path.

Reply:
yes, it would make sense to add an internal member function to compute this information.

Reply:
OK. I moved this into the new PR #11291.